testing/e2e/e2e_api_test.go

```diff
@@ -205,3 +253,271 @@ func (s *BeaconKitE2ESuite) TestValidatorsWithSpecificStatus() {
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4 million BERA
+		s.Require().True(balance <= 4e12, "Validator balance should not exceed 4 million BERA")
+	}
+}
+
+// TestValidatorBalancesMultipleIndices tests querying balances for multiple validator indices.
+func (s *BeaconKitE2ESuite) TestValidatorBalancesMultipleIndices() {
+	client := s.initBeaconTest()
+	indices := []phase0.ValidatorIndex{0, 1, 2}
+
+	balancesResp, err := client.ValidatorBalances(
+		s.Ctx(),
+		&beaconapi.ValidatorBalancesOpts{
+			State:   utils.StateIDHead,
+			Indices: indices,
+		},
+	)
+
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+	s.Require().Len(balancesResp.Data, len(indices))
+
+	// Verify all requested indices are present
+	returnedIndices := make(map[phase0.ValidatorIndex]bool)
+	for index, balance := range balancesResp.Data {
+		returnedIndices[index] = true
+		s.Require().True(balance > 0)
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4 million BERA
+		s.Require().True(balance <= 4e12, "Validator balance should not exceed 4 million BERA")
+	}
+	for _, idx := range indices {
+		s.Require().True(returnedIndices[idx], "Expected validator index not found in response")
+	}
+}
+
+// TestValidatorBalancesWithInvalidIndex tests querying validator balances with an invalid index.
+func (s *BeaconKitE2ESuite) TestValidatorBalancesWithInvalidIndex() {
+	client := s.initBeaconTest()
+
+	indices := []phase0.ValidatorIndex{999999} // Invalid index
+
+	balancesResp, err := client.ValidatorBalances(s.Ctx(), &beaconapi.ValidatorBalancesOpts{
+		State:   utils.StateIDHead,
+		Indices: indices,
+	})
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+	// Should return an empty list of balances
+	s.Require().Len(balancesResp.Data, 0)
+}
+
+// TestValidatorBalancesWithPubkey tests querying validator balances using a public key.
+func (s *BeaconKitE2ESuite) TestValidatorBalancesWithPubkey() {
+	client := s.initBeaconTest()
+
+	// First call validators to get the validator public key
+	validatorsResp, err := client.Validators(s.Ctx(), &beaconapi.ValidatorsOpts{
+		State: utils.StateIDHead,
+	})
+	s.Require().NoError(err)
+	s.Require().NotNil(validatorsResp)
+
+	validator := validatorsResp.Data[0]
+	s.Require().NotNil(validator)
+	pubkey := validator.Validator.PublicKey
+
+	balancesResp, err := client.ValidatorBalances(s.Ctx(), &beaconapi.ValidatorBalancesOpts{
+		State:   utils.StateIDHead,
+		Indices: []phase0.ValidatorIndex{},
+		PubKeys: []phase0.BLSPubKey{pubkey},
+	})
+	s.Require().NoError(err)
+	s.Require().NotNil(balancesResp)
+	s.Require().NotEmpty(balancesResp.Data)
+
+	// Verify balance data
+	for _, balance := range balancesResp.Data {
+		s.Require().NotNil(balance)
+		s.Require().True(balance > 0, "Validator balance should be positive")
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4 million BERA
+		s.Require().True(balance <= 4e12, "Validator balance should not exceed 4 million BERA")
```

Comment:
nit: 4e12 Gwei would be 4e3, or 4000, BERA

Reply:
my bad, Thank you! Addressed.